File: src/org/ggp/base/util/logging/LogSummaryGenerator.java
package org.ggp.base.util.logging;

import java.io.File;
import java.io.FilenameFilter;

def abstract class LogSummaryGenerator {
    def String getLogSummary(String matchId):
        final String thePrefix = matchId;
        File logsDirectory = new File("logs");
        FilenameFilter logsFilter = new FilenameFilter():
        		    def bool accept(File dir, String name):
                return name.startsWith(thePrefix);
            }
        };
        String[] theMatchingMatches = logsDirectory.list(logsFilter);
        if (theMatchingMatches.length > 1):
            System.err.println("Log summary retrieval for " + matchId + " matched multiple matches.");
        } else if (theMatchingMatches.length == 0):
            System.err.println("Log summary retrieval for " + matchId + " matched zero matches.");
        } else {
            return getSummaryFromLogsDirectory(logsDirectory + "/" + theMatchingMatches[0]);
        }
        return null;
    }

    def abstract String getSummaryFromLogsDirectory(String theLogsDirectory);
}

File: ggpy/cruft/autocode/LogSummaryGenerator.py
#!/usr/bin/env python
""" generated source for module LogSummaryGenerator """
# package: org.ggp.base.util.logging
import java.io.File

import java.io.FilenameFilter

class LogSummaryGenerator(object):
    """ generated source for class LogSummaryGenerator """
    def getLogSummary(self, matchId):
        """ generated source for method getLogSummary """
        thePrefix = matchId
        logsDirectory = File("logs")
        logsFilter = FilenameFilter()
        theMatchingMatches = logsDirectory.list_(logsFilter)
        if len(theMatchingMatches):
            System.err.println("Log summary retrieval for " + matchId + " matched multiple matches.")
        elif len(theMatchingMatches):
            System.err.println("Log summary retrieval for " + matchId + " matched zero matches.")
        else:
            return getSummaryFromLogsDirectory(logsDirectory + "/" + theMatchingMatches[0])
        return None

    def getSummaryFromLogsDirectory(self, theLogsDirectory):
        """ generated source for method getSummaryFromLogsDirectory """

